packages/next/src/utils.ts

```diff
@@ -1254,21 +1254,38 @@ async function getSourceFilePathFromPage({
     const extensionless = fsPath.replace(path.extname(fsPath), '');
 
     for (const ext of extensionsToTry) {
-      fsPath = `${extensionless}.${ext}`;
+      // for appDir, we need to treat "index.js" as root-level "page.js"
+      fsPath =
```

Comment:
Should this use a helper method like `normalizeIndexOutput()`?

Reply:
The check requires multiple vars to be passed, so I think I would leave it as-is